crates/subspace-farmer/src/plotting.rs

```diff
@@ -11,16 +12,59 @@ use subspace_archiving::pre_genesis_data;
 use subspace_core_primitives::objects::{GlobalObject, PieceObject, PieceObjectMapping};
 use subspace_core_primitives::Sha256Hash;
 use subspace_solving::SubspaceCodec;
+use tokio::sync::mpsc;
+
+pub struct Plotting {
+    sender: Option<mpsc::Sender<()>>,
+}
+
+impl Plotting {
+    pub fn start(
+        plot: Plot,
+        commitments: Commitments,
+        object_mappings: ObjectMappings,
+        client: RpcClient,
+        identity: Identity,
+    ) -> Self {
+        let (sender, receiver) = mpsc::channel(1);
```

Comment:
`mpsc` is not needed here though, oneshot is enough. But I guess that is because you use it with `select` and didn't know you probably need to fuse it to work in that context.

Reply:
That's correct :)

Reply:
it seems tokio handles the fuse in the background. All I needed was an owned version wrapped in an Option